sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_json.py

```diff
@@ -0,0 +1,83 @@
+# ------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# -------------------------------------------------------------------------
+
+ESCAPE_CHAR = "\\"
+DOUBLE_FORWARD_SLASH_COMMENT = "//"
+HASH_COMMENT = "#"
+MULTI_LINE_COMMENT = "/*"
+END_MULTI_LINE_COMMENT = "*/"
+DOUBLE_QUOTE = '"'
+NEW_LINE = "\n"
+
+
+def _strip_ignore_string(text, i, length, result):
+    """
+    Helper function to strip comments within a string literal.
+    Handles both single and double quotes, and escaped quotes.
+    Returns the updated index and the result list.
+
+    :param text: The input text to process.
+    :type text: str
+    :param i: The current index in the text.
+    :type i: int
+    :param length: The total length of the text.
+    :type length: int
+    :param result: The list to accumulate the processed characters.
+    :type result: list[str]
+    :raises ValueError: If there is an unterminated string literal.
+    :return: A tuple containing the updated result list and the next index.
+    :rtype: tuple[list[str], int]
+    """
+    while i < length:
+        current_char = text[i]
+        result.append(current_char)
+        if current_char == DOUBLE_QUOTE:
+            # Check for escaped quote
+            esc_count = 0
+            j = i - 1
+            while j >= 0 and text[j] == ESCAPE_CHAR:
+                esc_count += 1
+                j -= 1
+            if esc_count % 2 == 0:
+                return result, i + 1
+        i += 1
+    raise ValueError("Unterminated string literal")
+
+
+def remove_json_comments(text):
+    """
+    Removes comments from a JSON file. Supports //, #, and /* ... */ comments.
+    Writes the cleaned JSON to output_path if provided, else returns as string.
+
+    :param text: The input JSON string with comments.
+    :type text: str
+    :raises ValueError: If there is an unterminated string literal.
+    :return: A string with comments removed.
+    :rtype: str
+    """
+    result = []
+    i = 0
+    length = len(text)
+    while i < length:
+        current_char = text[i]
+        if current_char == DOUBLE_QUOTE:
+            result.append(current_char)
+            i += 1
+            result, i = _strip_ignore_string(text, i, length, result)
```

Comment:
We re-assign the `result`  after appending the double quote. Is that expected?

Reply:
I just rewrote the whole section to just return the index of the end of the string. It makes it quite a bit more clear in my opinion. What do you think.

Reply:
This is clearer, thanks!